### DISCUSSION TITLE
How should I notify my extension users of an available update?

### DISCUSSION THREAD

**Discussion by motdotla** [microsoft/vscode-discussions#126] [Extension Development QnA] [answered]
What do you recommend as a solution to notifying my extension users of an available update - for example, a bug patch?

There doesn't appear to be a built-in mechanism. It looks like I will need to write code at runtime. I could have it check the current version in code (as a constant) and then compare that to a remote ajax call that returns the latest published version. If there is a new version, notify the user with a pop-up or toast or something.

Is that the approach you recommend?

**Comment by gjsjohnmurray** [comment, marked as answer]
After you publish the new version on Marketplace, users who have configured their VS Code to auto-update extensions will receive your update. Others will see a badge appear on the Extensions icon on their sidebar, indicating how many enabled extensions are outdated. Then when they filter the Extensions view to show Outdated extensions they will see your extension listed with an "Update to x.y.z" button.

**Reply by burkeholland** [reply]
@motdotla This item is over a week old and @gjsjohnmurray provided a good summary of the options. I'm marking as closed for now. Please feel free to re-open if this does not answer your question.
